mindsdb/integrations/handlers/mediawiki_handler/mediawiki_handler.py

```diff
@@ -64,7 +90,7 @@ def check_connection(self) -> StatusResponse:
             self.connect()
```

Comment:
If the user has provided the username and password, I think they should be validated at the time of executing the `CREATE DATABASE` query, rather than waiting until a query is run. We can do this in the `check_connection()` method. In your case, I think this can be accomplished by adding a call to your `login()` method within `check_connection()`.